PyTDF/Node.py

```diff
@@ -54,19 +54,16 @@ def _dfs(self, node, prev_object= None):
             self._dfs(n, node.value)
 
     
-    def _graph_prune(self, node=None):
-
-        if not node:
-            node = self._get_head()
+    def _graph_prune(self):
 
         children = []
 
-        for n in node.next_nodes:
-            if self._graph_prune(n):
+        for n in self.next_nodes:
+            if n._graph_prune():
                 children.append(n)
-        node.next_nodes = children
+        self.next_nodes = children
 
-        if not len(node.next_nodes) and (len(gc.get_referrers(node)) <= 3):
+        if (not self.next_nodes) and (len(gc.get_referrers(self)) <= 3):
```

Comment:
you don't really need these parentheses, but I guess it's a matter of personal preference

Reply:
Yes. It isn't required. I'll remove this in the next commit :+1: